itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java

```diff
@@ -170,6 +180,34 @@ private Path mkDir(DistributedFileSystem fs, String pathString)
     return PathBuilder.fullyQualifiedHDFSUri(path, fs);
   }
 
+  /**
+   * Create location pointed by the given path with respect to the external table location root on
+   * DFS.
+   * @param location
+   * @return the path string to the location created with respect to the DFS
+   * @throws IOException
+   * @throws SemanticException
+   */
+  public String createExternalLoc(String location) throws IOException, SemanticException {
+    String extLoc = extLocRoot + "/" + location;
+    DistributedFileSystem fs = miniDFSCluster.getFileSystem();
+    mkDir(fs, extLoc);
+    return extLoc;
+  }
+
+  /**
+   * Delete location pointed by the given path with respect to the external table location root on
+   * DFS.
+   * @param location
+   * @throws IOException
+   * @throws SemanticException
+   */
+  public void deleteExternalLoc(String location) throws IOException, SemanticException {
+    String extLoc = extLocRoot + "/" + location;
```

Comment:
The caller already passes the complete path. So, this append makes it incorrect path.

Reply:
Right. Thanks for pointing it out. Done. I have renamed the function as deleteLocation() since it deletes given location, whether external or not.